### ISSUE TITLE
It seems the evaluation might have some bugs

### ISSUE THREAD

**Issue by nikepupu** [askforalfred/alfred#92]
<img width="538" alt="Screen Shot 2021-07-05 at 11 37 59 PM" src="https://user-images.githubusercontent.com/10679688/124554094-6adebc80-ddea-11eb-924f-7f894c0fec3d.png">
<img width="789" alt="Screen Shot 2021-07-05 at 11 37 04 PM" src="https://user-images.githubusercontent.com/10679688/124554116-74682480-ddea-11eb-8a58-12369c7e653c.png">

This is returning False. However, it should be True. 
This is from : pick_clean_then_place_in_recep-AppleSliced-None-DiningTable-27/trial_T20190907_151802_277016

**Comment by nikepupu** [comment]
This is from subgoal evaluation. The sequence of actions from my model:
[('PutObject', 'sink'), ('ToggleObjectOn', 'faucet'), ('ToggleObjectOff', 'faucet'), ('PickupObject', 'apple'), ('<<seg>>', 'END')]

**Comment by nikepupu** [comment]
<img width="305" alt="Screen Shot 2021-07-05 at 11 52 47 PM" src="https://user-images.githubusercontent.com/10679688/124557454-62888080-ddee-11eb-9b45-1151e75dc7a5.png">
<img width="318" alt="Screen Shot 2021-07-05 at 11 53 41 PM" src="https://user-images.githubusercontent.com/10679688/124557459-63b9ad80-ddee-11eb-8632-b4cbda2f4289.png">
<img width="473" alt="Screen Shot 2021-07-05 at 11 53 53 PM" src="https://user-images.githubusercontent.com/10679688/124557468-66b49e00-ddee-11eb-9376-e6e7f7b89d70.png">
<img width="332" alt="Screen Shot 2021-07-05 at 11 54 00 PM" src="https://user-images.githubusercontent.com/10679688/124557475-687e6180-ddee-11eb-8f48-b88c84b9950a.png">
image sequence before each actions.

**Comment by MohitShridhar** [comment]
@nikepupu thanks for looking into this! 

The `cleaned` state is maintained [here](https://github.com/askforalfred/alfred/blob/master/env/thor_env.py#L165). Sounds like something weird is happening with  the `objectType` of sliced objects, and that's messing up [this](https://github.com/askforalfred/alfred/blob/b081a5cc6b400f02ab7218004751425848fc8bc1/gen/utils/game_util.py#L295).

Are you using `ai2thor==2.1.0`?

**Comment by nikepupu** [comment]
Just checked again. I am using ai2thor==2.1.0. Thanks for the quick reply.

**Comment by MohitShridhar** [comment]
@nikepupu I'll look into this later this week when I get some time.

For the time being, I would suggest putting a breakpoint [here](https://github.com/askforalfred/alfred/blob/b081a5cc6b400f02ab7218004751425848fc8bc1/gen/utils/game_util.py#L295) and checking if the `objectType` is somehow wrong.

**Comment by nikepupu** [comment]
Some additional information for a related issue:
<img width="921" alt="Screen Shot 2021-07-07 at 3 14 28 PM" src="https://user-images.githubusercontent.com/10679688/124835924-f2d1dd00-df36-11eb-8583-348b2cf29274.png">
<img width="328" alt="Screen Shot 2021-07-07 at 3 14 36 PM" src="https://user-images.githubusercontent.com/10679688/124835937-f6656400-df36-11eb-9ebd-e291214feb60.png">

I think a better approach would be comparing objectName's prefix and then compare objects states. Issliced, ispickedup, etc... 
objectId seems to be unreliable.

**Comment by nikepupu** [comment]
this is a race condition 
pick_and_place_simple-Knife-None-SideTable-3/trial_T20190918_184236_557252
<img width="233" alt="Screen Shot 2021-07-07 at 4 05 28 PM" src="https://user-images.githubusercontent.com/10679688/124839172-2a438800-df3d-11eb-88aa-5a0b6671cabe.png">

**Comment by MohitShridhar** [comment]
Thanks @nikepupu! Fixed in https://github.com/askforalfred/alfred/commit/904e1b025a626d201123a898f837e4edfeea741e.

The specific slice doesn't matter, but the objectId might be still relevant for some referring expressions e.g. "the apple on the left". So I added a simple fix to remove `Sliced...` from the ID string.

Fortunately, this reward function is not used anywhere in the leaderboard evaluation script, so existing and future submissions are unaffected.
